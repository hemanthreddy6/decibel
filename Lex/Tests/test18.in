function scaleWave <- (factor: float, t: float): float {
    return SIN(t) * factor;
}

function halfScale <- scaleWave(0.5, _);
softWave: audio <- audio(halfScale, 440hz, 5s);

play softWave;
save softWave -> "soft_wave.wav";
