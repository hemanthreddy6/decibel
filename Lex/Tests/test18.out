FUNCTION
IDENTIFIER
<-
(
IDENTIFIER
:
FLOAT
,
IDENTIFIER
:
FLOAT
)
:
FLOAT
{
RETURN
SIN
(
IDENTIFIER
)
*
IDENTIFIER
;
}
FUNCTION
IDENTIFIER
<-
IDENTIFIER
(
FLOAT_LITERAL
,
_
)
;
AUDIO
IDENTIFIER
<-
AUDIO
(
IDENTIFIER
,
FLOAT_LITERAL
,
FLOAT_LITERAL
)
;
PLAY
IDENTIFIER
;
SAVE
IDENTIFIER
->
STRING_LITERAL
;
